File: output/CodeLlama/codenet/C++/Python/s314711832.py
#include <iostream>
#include <vector>
#define MIN(a, b) ((a < b) ? a : b)
#define INF 1000000001
#define N 100000

using namespace std;

struct edge {
  int source;
  int target;
  int weight;
};

vector<edge> edges;

int distances[N];
int prev_distances[N] = {0};



int main() {
  int V, E, r;

  cin >> V >> E >> r;

  for (int i = 0; i < E; i++) {
    int source, target, distance;
    cin >> source >> target >> distance;
    edge new_edge = { source, target, distance};
    edges.push_back(new_edge);
    
  }

  //initialize distances array
 
  for (int i = 0; i < N; i++) {
    distances[i] = INF ; //max value
  }

  distances[r] = 0;

 
  for (int i = 1; i < V; i++) {
    for (int j=0; j< V; j++) {
      prev_distances[j] = distances[j];  
    }
    
    for (vector<edge>::iterator edgeptr = edges.begin(); edgeptr != edges.end(); edgeptr++) {
      distances[edgeptr->target] = MIN(distances[edgeptr->source] + edgeptr->weight, distances[edgeptr->target]);
    
    }

    bool same = True;
    for (int j=0; j< V; j++) {
      if (distances[j] != prev_distances[j]):
	same = False;
	break;
    }

    if (same):
      break;
  }

  for (int i = 0; i < V; i++) {
    if (distances[i] == INF):
      print("INF")
    else:
      print(distances[i])
  }
}

File: output/CodeLlama/codenet/C++/Java/s314711832.java
import java.util.Scanner;
import java.util.Vector;

public class s314711832 {

  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int V = sc.nextInt();
    int E = sc.nextInt();
    int r = sc.nextInt();

    Vector<Edge> edges = new Vector<Edge>();

    for (int i = 0; i < E; i++) {
      int source = sc.nextInt();
      int target = sc.nextInt();
      int weight = sc.nextInt();
      Edge new_edge = new Edge(source, target, weight);
      edges.add(new_edge);
    }

    int[] distances = new int[V];
    int[] prev_distances = new int[V];

    for (int i = 0; i < V; i++) {
      distances[i] = Integer.MAX_VALUE;
    }

    distances[r] = 0;

    for (int i = 1; i < V; i++) {
      for (int j = 0; j < V; j++) {
        prev_distances[j] = distances[j];
      }

      for (int j = 0; j < E; j++) {
        Edge edge = edges.get(j);
        distances[edge.target] = Math.min(distances[edge.source] + edge.weight, distances[edge.target]);
      }

      boolean same = true;
      for (int j = 0; j < V; j++) {
        if (distances[j] != prev_distances[j]) {
          same = false;
          break;
        }
      }

      if (same) break;
    }

    for (int i = 0; i < V; i++) {
      if (distances[i] == Integer.MAX_VALUE) {
        System.out.println("INF");
      } else {
        System.out.println(distances[i]);
      }
    }
  }
}

class Edge {
  int source;
  int target;
  int weight;

  public Edge(int source, int target, int weight) {
    this.source = source;
    this.target = target;
    this.weight = weight;
  }
}